lib/chef/provider/package/yum.rb

```diff
@@ -1081,22 +1082,24 @@ def load_current_resource
                 @new_resource.version($2)
               end
             end
-          end
-
-          if @new_resource.version
-            new_resource = "#{@new_resource.package_name}-#{@new_resource.version}#{yum_arch}"
+            @candidate_version << @new_resource.version
+            installed_version << @yum.installed_version(@current_resource.package_name, arch)
```

Comment:
`load_current_resource` probably makes more sense. The one downside there is none of the providers call super, so fixing requires touching all the package providers.

Why is doing it in the constructor rude in this case? It seems like you shouldn't let anything run if the code has a "syntax" error